app/services/ahoy_queries.rb

```diff
@@ -34,19 +26,183 @@ def interval_by_date_range(date_range)
     when YESTERDAY
       (Date.yesterday.to_time(:utc).beginning_of_day..yesterday_end)
     when LAST_7_DAYS
-      ((today - 7.days)..today.end_of_day)
+      ((today - 8.days)..yesterday_end)
     when LAST_30_DAYS
-      ((today - 30.days)..today.end_of_day)
+      ((today - 31.days)..yesterday_end)
     when LAST_MONTH
       last_month = today.prev_month
       (last_month.beginning_of_month..last_month.end_of_month)
     when LAST_QUARTER
       prev_quarter = today.prev_quarter
       (prev_quarter.beginning_of_quarter..prev_quarter.end_of_quarter)
     when LAST_12_MONTHS
-      ((today.prev_year)..today.end_of_day)
+      ((today.prev_year)..yesterday_end)
     else
       #not valid date range
     end
   end
+
+
+  def get_number_of_updates_last_thirty_days(location_id)
+    Ahoy::Event
```

Comment:
I winder if we could use the Ahoy Queries module here instead of making the query directly here?